distributed/comm/core.py

```diff
@@ -238,7 +238,7 @@ def resolve_address(addr):
 
 
 @gen.coroutine
-def connect(addr, timeout=3, deserialize=True):
+def connect(addr, timeout=3, deserialize=True, ssl_options=None):
```

Comment:
I think this is the wrong approach. SSL will not be applicable to all transports. Instead, I think SSL (I'd rather call it TLS btw) should be exposed as a distinct transport from TCP.